modules/monitoring-example-remote-write-queue-configuration.adoc

```diff
@@ -0,0 +1,68 @@
+// Module included in the following assemblies:
+//
+// * observability/monitoring/configuring-the-monitoring-stack.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="example-remote-write-queue-configuration_{context}"]
+= Example remote write queue configuration
+
+// Set attributes to distinguish between cluster monitoring examples and user workload monitoring examples.
+ifndef::openshift-dedicated,openshift-rosa[]
+:configmap-name: cluster-monitoring-config
+:namespace-name: openshift-monitoring
+:prometheus-instance: prometheusK8s
+endif::openshift-dedicated,openshift-rosa[]
+ifdef::openshift-dedicated,openshift-rosa[]
+:configmap-name: user-workload-monitoring-config
+:namespace-name: openshift-user-workload-monitoring
+:prometheus-instance: prometheus
+endif::openshift-dedicated,openshift-rosa[]
+
+You can use the `queueConfig` object for remote write to tune the remote write queue parameters. The following example shows the queue parameters with their default values for 
+ifndef::openshift-dedicated,openshift-rosa[]
+default platform monitoring
+endif::openshift-dedicated,openshift-rosa[]
+ifdef::openshift-dedicated,openshift-rosa[]
+monitoring for user-defined projects
+endif::openshift-dedicated,openshift-rosa[]
+in the `{namespace-name}` namespace.
+
+.Example configuration of remote write parameters with default values
+[source,yaml,subs="attributes+"]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {configmap-name}
+  namespace: {namespace-name}
+data:
+  config.yaml: |
+    {prometheus-instance}:
+      remoteWrite:
+      - url: "https://remote-write-endpoint.example.com" 
+        <endpoint_authentication_credentials>
+        queueConfig:
+          capacity: 10000 #<1>
+          minShards: 1 #<2>
+          maxShards: 50 #<3>
+          maxSamplesPerSend: 2000 #<4>
+          batchSendDeadline: 5s #<5>
+          minBackoff: 30ms #<6>
+          maxBackoff: 5s #<7>
+          retryOnRateLimit: false #<8>
+          sampleAgeLimit: 0s #<9>
+----
+<1> The number of samples to buffer per shard before they are dropped from the queue.
+<2> The minimum number of shards.
+<3> The maximum number of shards.
+<4> The maximum number of samples per send.
+<5> The maximum time for a sample to wait in buffer.
+<6> The initial time to wait before retrying a failed request. It gets doubled for every retry up to the `max_backoff` time.
+<7> The maximum time to wait before retrying a failed request.
+<8> Set this parameter to `true` if you want to retry a request after receiving a 429 status code from the remote write storage.
```

Comment:
do we need to mention it's an *experimental feature*?
```
# oc explain prometheus.spec.remoteWrite.queueConfig.retryOnRateLimit
GROUP:      monitoring.coreos.com
KIND:       Prometheus
VERSION:    v1

FIELD: retryOnRateLimit <boolean>

DESCRIPTION:
    Retry upon receiving a 429 status code from the remote-write storage.
    
    
    This is an *experimental feature*, it may change in any upcoming release
    in a breaking way.
```

Reply:
@simonpasquier would we concider this a TP feature? Otherwise, I would probably avoid including this one.

Reply:
the description comes from the upstream CRD which itself mimics the upstream Prometheus documentation. I have no good answer for this but in practice:
* It's unlikely that the feature goes away.
* Even if it was, we would continue supporting it in the CMO configmap.